templates/cc/msg.go

```diff
@@ -48,10 +48,30 @@ const msgTpl = `
 	{{ end }}{{ end }}
 	{{ end }}{{ end }}
 
-	{{ if has .Rules "Pattern"}}{{ if .Rules.Pattern }}
-	{{/* TODO(akonradi) implement pattern matching
-		var {{ lookup .Field "Pattern" }} = regexp.MustCompile({{ lit .Rules.GetPattern }})
-	*/}}
+        {{ if has .Rules "Pattern"}}{{ if .Rules.Pattern }}
+               const re2::RE2 {{ lookup .Field "Pattern" }}(std::string({{ lit .Rules.GetPattern }},
+                                              sizeof({{ lit .Rules.GetPattern }}) - 1), re2::RE2::Quiet);
+	{{ end }}{{ end }}
+
+	{{ if has .Rules "Items"}}{{ if .Rules.Items }}
+        {{ if has .Rules.Items.GetString_ "Pattern" }} {{ if .Rules.Items.GetString_.Pattern }}
+               const re2::RE2 {{ lookup .Field "Pattern" }}(std::string({{ lit .Rules.Items.GetString_.GetPattern }},
```

Comment:
The construction of a std::string here doesn't seem terrible; I don't think there's a better way to deal with null-terminated strings unless you want to introduce absl/std::string_view. That would save a copy on startup.